File: solution/day23/src/test/java/org/craftedsw/tripservicekata/trip/TripRepositoryShould.java
package org.craftedsw.tripservicekata.trip;

import org.craftedsw.tripservicekata.exception.CollaboratorCallException;
import org.craftedsw.tripservicekata.user.UserBuilder;
import org.junit.jupiter.api.DisplayNameGeneration;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThatThrownBy;
import static org.junit.jupiter.api.DisplayNameGenerator.ReplaceUnderscores;

@DisplayNameGeneration(ReplaceUnderscores.class)
public class TripRepositoryShould {
    private final TripRepository tripRepository = new TripRepository();

    @Test
    void retrieve_user_trips_throw_an_exception() {
        assertThatThrownBy(() -> tripRepository.findTripsByUser(UserBuilder.aUser().build()))
                .isInstanceOf(CollaboratorCallException.class);
    }
}

File: solution/c#/Day23/Day23.Tests/Trips/TripRepositoryShould.cs
using Day23.Exception;
using Day23.Tests.Users;
using Day23.Trips;
using FluentAssertions;
using Xunit;

namespace Day23.Tests.Trips
{
    public class TripRepositoryShould
    {
        private readonly ITripRepository _tripRepository = new TripRepository();

        [Fact]
        public void Retrieve_User_Trips_Throw_An_Exception()
        {
            var findTrips = () => _tripRepository.FindTripsByUser(UserBuilder.AUser().Build());
            findTrips.Should().Throw<CollaboratorCallException>();
        }

        private class TripRepository : ITripRepository;
    }
}